test/libsolidity/syntaxTests/array/call_uninitialized_function_through_array.sol

```diff
@@ -0,0 +1,16 @@
+// this used to be a semantics tests that was expected to fail, but is now caught at compile-time
+contract C {
+    int mutex;
+    function t() returns (uint) {
+        if (mutex > 0)
+            { assembly { mstore(0, 7) return(0, 0x20) } }
+        mutex = 1;
+        // Avoid re-executing this function if we jump somewhere.
+        function() internal returns (uint)[200] x;
```

Comment:
Oh wow, actually - why does this fail? Would something like `uint[3] memory x; return x[2];` also fail? It should not...

Reply:
Ah sorry, it is not `memory` but `storage`, I see.

Reply:
Yeah, right. I *think* I wanted to use a memory array here and not a storage array. The idea was not to test uninitialized storage array pointers, but uninitialized function pointers.